.lintrunner.toml

```diff
@@ -159,6 +159,34 @@ command = [
 ]
 is_formatter = true
 
+[[linter]]
+code = 'irange'
+include_patterns = [
+    '**/*.h',
+    '**/*.cpp',
+    '**/*.cu',
+    '**/*.inl',
+]
+exclude_patterns = [
+    'pytorch/**/*.cpp',
+    'third_party/**',
+]
+command = [
+    'python3',
+    'tools/linter/adapters/grep_linter.py',
+    '--pattern=irange',
+    '--linter-name=irange',
+    '--error-name=using c10::irange, use std::views::iota instead',
+    '--replace-pattern=s/irange/std::views::iota/',
```

Comment:
Should the pattern be `c10::irange` instead?

Reply:
I also want to match `#include <c10/util/irange.h>`, currently, it will be replaced as `#include <c10/util/std::views::iota.h>`, this is incorrect, but informative enough. 😂

Reply:
This sounds like a bug 😂 

Can we instead have two patterns, one from c10::irange to std::views::iota.h and the other from `<c10/util/irange.h>` to whatever includes `iota_view`? If that's not good enough, I think we can always choose not to replace and leave that to the user. Sometimes, this is better than doing the wrong magic.

Reply:
Changed the linter to replace `c10::irange` and the header separately. I generally don't like having too many linters like this, but looks like if we create a `nvfuser::irange`, this is the only way.